addons/mail/models/discuss/discuss_channel.py

```diff
@@ -904,6 +904,12 @@ def _to_store(self, store: Store):
                 member_of_current_user_by_channel[member.channel_id] = member
         for channel in self:
             member = member_of_current_user_by_channel.get(channel, self.env['discuss.channel.member']).with_prefetch([m.id for m in member_of_current_user_by_channel.values()])
+            self_member_data = member._discuss_channel_member_format(extra_fields={
+                "last_interest_dt": True,
+                "message_unread_counter": True,
+                "message_unread_counter_bus_id": bus_last_id,
+                "new_message_separator": True
+            }).get(member)
```

Comment:
I think this is likely the cause of linear increase of query count: if you ensure to fetch and format self members for all channels at once rather than each time in the loop, this would make code more query-efficient